### PR TITLE
7903121: initial jextract tool code, tests and samples

### PR DESCRIPTION
**Pull request by sundararajana**
Adding initial standalone jextract code, tests and samples.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [CODETOOLS-7903121](https://bugs.openjdk.java.net/browse/CODETOOLS-7903121): initial jextract tool code, tests and samples

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer)
 * [Jorn Vernee](https://openjdk.java.net/census#jvernee) (@JornVernee - no project role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jextract pull/2/head:pull/2` \
`$ git checkout pull/2`

Update a local copy of the PR: \
`$ git checkout pull/2` \
`$ git pull https://git.openjdk.java.net/jextract pull/2/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2`

View PR using the GUI difftool: \
`$ git pr show -t 2`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jextract/pull/2.diff">https://git.openjdk.java.net/jextract/pull/2.diff</a>

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,3 +153,40 @@ It is easy to see how this mechanism allows developers to look into the set of s
 
 For more examples on how to use the `jextract` tool with real-world libraries, please refer to this [document](https://github.com/openjdk/panama-foreign/blob/d8c0fe5918cb1c6c744eb26797ea4fa04142c237/doc/panama_jextract.md).
 
+
+#### Building jextract tool
```
Comment (mcimadamore) [inline README.md:157]: You probably want just `###` in these sections (for uniformity)

File: README.md [hunk 2]
```diff
@@ -155,3 +153,40 @@ It is easy to see how this mechanism allows developers to look into the set of s
...
+
+```sh
+
+$ sh ./gradlew -Pjdk18_home=<jdk18_home_dir> -PLIBCLANG_HOME=<libclang_dir> -Pjtreg_home=<jtreg_dir> clean jtreg
+
+```
+
+### jextract samples
+
+jextract samples can be found "samples" top-level directory. Building/running particular sample may require
```
Comment (mcimadamore) [inline README.md:191]: This should probably go somewhere more toplevel - e.g. in the first intro - but I'll do another pass on the document after you push.

Comment (mcimadamore) [inline README.md:191]: This should probably go somewhere more toplevel - e.g. in the first intro - but I'll do another pass on the document after you push.

File: README.md [hunk 3]
```diff
@@ -155,3 +153,40 @@ It is easy to see how this mechanism allows developers to look into the set of s
...
+jdk 17 or below. Or PATH should contain java from jdk 17 or below. jdk18 build is
+needed to build jextract which is passed from command with -Pjdk18_home option.
+
+You can download jdk18 early access build from [https://jdk.java.net/18/](https://jdk.java.net/18/)
+
+For Windows, please use gradlew.bat.
+
+```sh
+
+$ sh ./gradlew -Pjdk18_home=<jdk18_home_dir> -PLIBCLANG_HOME=<libclang_dir> clean verify
```
Comment (mcimadamore) [inline README.md:172]: I think the clang property should be lowercase, for consistency with other properties

Comment (mcimadamore) [inline README.md:172]: I think the clang property should be lowercase, for consistency with other properties

**Review by mcimadamore — COMMENTED** [review]
Looks good. I've tested the build on linux-x64 and it works ok. I've added some minor comments to make things a bit more consistent.

File: README.md (hunk 2 above)
Comment (sundararajana) [reply README.md:191]: sounds good.

File: README.md (hunk 1 above)
Comment (sundararajana) [reply README.md]: Fixed that.

File: README.md (hunk 2 above)
Comment (sundararajana) [reply README.md:191]: sounds good

**Review by JornVernee — APPROVED** [review]
Works on Windows

**Pull request closed without merging**